src/raster/buffer.rs

```diff
@@ -0,0 +1,86 @@
+use crate::raster::GdalType;
+
+#[cfg(feature = "ndarray")]
+use ndarray::Array2;
+
+/// A 2-D array backed by it's `size` (cols, rows) and a row-major `Vec<T>` and it's dimensions.
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
Should we also derive `Eq`, `ParialOrd` and `Ord`?

Reply:
`Eq` and `Ord` get really messy with `Buffer<f32|f64>`.

Reply:
Just a little. `#[derive]` [automatically adds](https://github.com/rust-lang/rust/issues/26925) a `T: Eq` bound, which works in our favour this time.